Made The Code Clearer

ranbir7 opened the pull request:
Codes and lines were made smaller to make it _readable_ and some type hints are added.

ranbir7 committed:
Made the Code Clearer

ranbir7 committed:
Made the code clearer

ranbir7 committed:
Made the code clearer & added Type Hints.

ranbir7 committed:
Update image_object.py

ranbir7 commented on autoprof/image/image_object.py:
```diff
@@ -107,40 +136,60 @@ def shape(self):
         self.set_data(data)
         
-    def set_data(self, data, require_shape = True):
+    def set_data(self, data: Union[torch.Tensor, np.ndarray], require_shape: bool = True):
```
is the bool here making a problem?

ConnorStoneAstro replied:
The `bool` looks ok to me

ConnorStoneAstro commented on autoprof/image/image_object.py:
```diff
@@ -67,29 +65,60 @@ def __init__(self,
+
+        #set Zeropoint
+        if zeropoint is None:
+            self.zeropoint = none
```
should be `self.zeropoint = None`

ConnorStoneAstro commented on autoprof/image/image_object.py:
```diff
@@ -67,29 +65,60 @@ def __init__(self,
-                self.pixelscale = torch.as_tensor(pixelscale, dtype = AP_config.ap_dtype, device = AP_config.ap_device)
+                self.pixelscale = torch.as_tensor(pixelscale, 
+                                                 dtype = AP_config.ap_dtype, 
+                                                 device = AP_config.ap_device)
```
I think an extra space is needed here for the indentation

ConnorStoneAstro commented on autoprof/image/image_object.py:
```diff
@@ -107,40 +136,60 @@ def shape(self):
         """
-        return torch.isclose(((self.center - self.origin) / self.pixelscale) % 1, torch.tensor(0.5, dtype = AP_config.ap_dtype, device = AP_config.ap_device), atol = 0.25)
+        return torch.isclose(((self.center - self.origin) / self.pixelscale) % 1, 
+                            0.5, 
```
Looks like this line got mixed up. See the previous version, the `0.5` was wrapped in `torch.tensor` along wiht some other adjustments

ConnorStoneAstro commented on autoprof/image/image_object.py:
```diff
@@ -107,40 +136,60 @@ def shape(self):
-        self._data = data.to(dtype = AP_config.ap_dtype, device = AP_config.ap_device) if isinstance(data, torch.Tensor) else torch.as_tensor(data, dtype = AP_config.ap_dtype, device = AP_config.ap_device)
-        
+        if isinstance(data, torch.Tensor):
+            self.data = data.to(dtype = AP_config.ap_dtype, device=AP_config.ap_device)
```
This is where the recursion is happening. Be careful to use `self._data` not `self.data` when setting the value

ConnorStoneAstro commented on autoprof/image/image_object.py:
```diff
@@ -107,40 +136,60 @@ def shape(self):
+        if isinstance(data, torch.Tensor):
+            self.data = data.to(dtype = AP_config.ap_dtype, device=AP_config.ap_device)
+        else:
+            self.data = torch.as_tensor(data, dtype = AP_config.ap_dtype, device = AP_config.ap_device)
```
Same here, should be `self._data` at the beginning

ConnorStoneAstro reviewed (commented):
These are some great updates! I found where the recursion error is happening. Try fixing these points and lets see if it works!

ranbir7 committed:
Update image_object.py

ranbir7 committed:
Update image_object.py

ranbir7 committed:
Update image_object.py

ranbir7 committed:
Update image_object.py

ConnorStoneAstro commented on autoprof/image/image_object.py:
```diff
@@ -107,40 +136,56 @@ def shape(self):
 
         """
-        return torch.isclose(((self.center - self.origin) / self.pixelscale) % 1, torch.tensor(0.5, dtype = AP_config.ap_dtype, device = AP_config.ap_device), atol = 0.25)
+        return torch.isclose(((self.center - self.origin) / self.pixelscale) % 1, torch.Tensor(0.5, atol = 0.25, dtype = AP_config.ap_dtype, device = AP_config.ap_device))
```
Looks like the `atol = 0.25` argument has been put inside the tensor creation. It should come after the tensor is created

ranbir7 committed:
Updated image_object.py

ranbir7 committed:
Update image_object.py

the torch.tensor was creating problems 
I kept it as torch._T_ensor 
whereas I should have kept the _T_ small..

ConnorStoneAstro reviewed (approved):
Everything looks great!
